example/lib/example/shoehorn_handler.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Example.ShoehornHandler do
+  use Shoehorn.Handler
+
+  def init(_opts) do
+    {:ok, %{restart_counts: 0}}
+  end
+
+  def application_started(app, s) do
+    IO.puts "Application started: #{inspect app}"
+    {:continue, s}
+  end
+
+  def application_exited(app, _reason, %{restart_counts: restart_counts} = state)
+      when restart_counts < 2 do
+    IO.puts("Application stopped: #{inspect(app)} #{inspect(state)}")
+    Application.ensure_all_started(app)
+    {:continue, %{state | restart_counts: restart_counts + 1}}
+  end
+
+  def application_exited(app, _reason, %{restart_counts: restart_counts} = state)
+      when restart_counts < 4 do
```

Comment:
I'm intrigued by splitting out the < 2 and <4 cases. Is this intentional? I don't see any differences.

Reply:
Originally we had halt, continue and restart. We should remove one or the other now that we have halt and continue.